### PR TITLE
Add documentation for BackendTLSPolicy

### LINKED ISSUE #2483: Documentation for BackendTLSPolicy
**What would you like to be added**:

- [x] Update https://gateway-api.sigs.k8s.io/guides/tls/ with BackendTLSPolicy

If time permits, 
- [x] Add API type BackendTLSPolicy

**Why this is needed**:
Needed to document new feature in release 1.0

### PR DESCRIPTION
**Pull request by candita**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
Add documentation for v1alpha2 BackendTLSPolicy.

**Which issue(s) this PR fixes**:

Fixes #2483

**Does this PR introduce a user-facing change?**:

```release-note
Documentation added for v1alpha2 BackendTLSPolicy.
```

### CODE REVIEW

File: examples/experimental/v1alpha2/backendtlspolicy-ca-certs.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#$ Used in:
+#$ - site-src/guides/tls.md
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: BackendTLSPolicy
+metadata:
+  name: tls-upstream-auth
+spec:
+  targetRef: auth-service
```
Comment (youngnick) [inline examples/experimental/v1alpha2/backendtlspolicy-ca-certs.yaml:8]: These are making the apply tests fail because they need to be a fully qualified reference:
```suggestion
  targetRef:
    group: ""
    kind: Service
    name: auth-service
```

**Comment by youngnick** [comment]
Aside from the CI passing, this LGTM. I'll approve and leave another read for someone else.

/approve

File: examples/experimental/v1alpha2/backendtlspolicy-ca-certs.yaml (hunk 1 above)
Comment (candita) [reply examples/experimental/v1alpha2/backendtlspolicy-ca-certs.yaml]: Fixed.  Thanks Nick.

Comment (candita) [reply examples/experimental/v1alpha2/backendtlspolicy-ca-certs.yaml]: Fixed again, thanks for the help!

**Comment by candita** [comment]
@robscott @shaneutt @bowei could you ptal and lgtm?  I will be posting another PR for the second task in https://github.com/kubernetes-sigs/gateway-api/issues/2483

File: site-src/guides/tls.md [hunk 2]
```diff
@@ -25,30 +25,33 @@ upstream connections is managed independently.
 
 Depending on the Listener Protocol, different TLS modes and Route types are supported.
 
-Listener Protocol | TLS Mode | Route Type Supported
---- | --- | ---
-TLS | Passthrough | TLSRoute
-TLS | Terminate | TCPRoute
-HTTPS | Terminate | HTTPRoute
+| Listener Protocol | TLS Mode    | Connection/s Affected   | Route Type Supported |
```
Comment (robscott) [inline site-src/guides/tls.md:28]:
```suggestion
| Listener Protocol | TLS Mode    | Connections Affected   | Route Type Supported |
```

File: site-src/guides/tls.md [hunk 3]
```diff
@@ -25,30 +25,33 @@ upstream connections is managed independently.
...
+|-------------------|-------------|-------------------------|----------------------|
+| TLS               | Passthrough | Downstream and Upstream | TLSRoute             |
+| TLS               | Terminate   | Downstream              | TCPRoute             |
+| HTTPS             | Terminate   | Downstream              | HTTPRoute            |
+| HTTPS             | Backend     | Upstream                | HTTPRoute            |
```
Comment (robscott) [inline site-src/guides/tls.md:33]: I think the concepts of "downstream" and "upstream" are going to be very confusing here. I'd recommend copying the diagram you made for the GEP and adding it here along with the corresponding terminology. https://gateway-api.sigs.k8s.io/geps/gep-1897/#overview-what-do-we-want-to-do

File: examples/experimental/v1alpha2/backendtlspolicy-ca-certs.yaml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  targetRef:
+    kind: Service
+    name: auth-service
+    group: ""
+  tls:
+    caCertRefs:
+      - kind: ConfigMapReference
+        name: auth-cert
+        group: ""
+    hostname: auth-pod.example.com
```
Comment (robscott) [inline examples/experimental/v1alpha2/backendtlspolicy-ca-certs.yaml:17]: Would avoid using "pod" here as it could lead to confusion
```suggestion
    hostname: auth.example.com
```

File: examples/experimental/v1alpha2/backendtlspolicy-system-certs.yaml [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+metadata:
+  name: tls-upstream-dev
+spec:
+  targetRef:
+    kind: Service
+    name: dev-service
+    group: ""
+  tls:
+    wellKnownCACerts: "System"
+    hostname: dev-pod.example.com
```
Comment (robscott) [inline examples/experimental/v1alpha2/backendtlspolicy-system-certs.yaml:14]: Same comment as above
```suggestion
    hostname: dev.example.com
```

File: site-src/guides/tls.md [hunk 6]
```diff
@@ -98,6 +101,56 @@ would be invalid.
...
+
+The `TargetRef` identifies the `Service` for which your HTTPRoute requires TLS.
+The `TLS` configuration for upstream TLS contains a required `Hostname`, and either
+`CACertRefs` or `WellKnownCACerts`.
+`Hostname` refers to the SNI the Gateway should use to connect to the backend, and
+must match the certificate served by the backend pod.
+`CACertRefs` refer to one or more PEM-encoded TLS certificates.
+If there are no specific certificates to use, then you must set `WellKnownCACerts` to
+"System" to tell the Gateway to use a set of trusted CA Certificates.  Refer to your
+implementation documentation for more details.
```
Comment (robscott) [inline site-src/guides/tls.md:122]:
```suggestion
"System" to tell the Gateway to use a set of trusted CA Certificates. There may be some
variation in which system certificates are used by each implementation. Refer to documentation
from your implementation of choice for more information.
```

File: site-src/guides/tls.md [hunk 7]
```diff
@@ -98,6 +101,56 @@ would be invalid.
...
+    - Wildcard hostnames are not allowed.
+
+### Examples
+
+#### Using System Certificates
+
+In this example, the `BackendTLSPolicy` is configured to use system certificates
+to connect with a TLS-encrypted upstream connection to the pod called `dev-pod`,
+with hostname `dev-pod.example.com`, and
+served by the service called `dev-service`.
```
Comment (robscott) [inline site-src/guides/tls.md:136]: Similar to above, would avoid suggesting that this refers to a single Service.
```suggestion
to connect with a TLS-encrypted upstream connection where Pods backing the
`dev` Service are expected to serve a valid certificate for `dev.example.com`.
```

File: site-src/guides/tls.md [hunk 8]
```diff
@@ -98,6 +101,56 @@ would be invalid.
...
+{% include 'experimental/v1alpha2/backendtlspolicy-system-certs.yaml' %}
+```
+
+#### Using Explicit CA Certificates
+
+In this example, the `BackendTLSPolicy` is configured to use certificates defined
+in the configuration map `auth-cert`
+to connect with a TLS-encrypted upstream connection to the pod called `auth-pod`,
+with hostname `auth-pod.example.com`, and
+served by the service called `auth-service`.
```
Comment (robscott) [inline site-src/guides/tls.md:148]: Would recommend a similar set of changes for this example.

**Review by robscott — COMMENTED** [review]
Thanks for the work on this @candita!

File: site-src/guides/tls.md (hunk 3 above)
Comment (candita) [reply site-src/guides/tls.md:33]: @robscott the concepts of downstream and upstream pre-exist in this document and the diagram.  I added a section called "Upstream", but there was already a section called "Downstream".    I can change "upstream" to "backend" here, but then what do I call what was formerly called "upstream" and not referencing the changes this introduces?

File: site-src/guides/tls.md (hunk 7 above)
Comment (candita) [reply site-src/guides/tls.md:137]: I think you meant "to avoid suggesting that this refers to a single Pod"?  Agreed.

File: site-src/guides/tls.md (hunk 3 above)
Comment (candita) [reply site-src/guides/tls.md:33]: Related comment, if I change "upstream" to "backend", shouldn't i also change "downstream" to "frontend'? Mixing the metaphors "downstream" and "backend" doesn't seem less confusing.

Comment (robscott) [reply site-src/guides/tls.md:33]: 🤦 you're right, I was not looking at the full deploy preview and related context, in which case the existing terminology you've used makes sense. 

I do really like the diagram you created for the GEP though, so if you wanted to make a version that uses the terminology in this doc (upstream and downstream), that could be a great upgrade on the current diagram here. (That's a non-blocking/optional suggestion though).

On further inspection, I would actually revert the change to this table. While "Passthrough" and "Terminate" are options on Listener TLS, "Backend" is not and is somewhat orthogonal.

File: site-src/guides/tls.md [hunk 9]
```diff
@@ -25,30 +25,33 @@ upstream connections is managed independently.
...
 
 Please note that in case of `Passthrough` TLS mode, no TLS settings take
-effect as the TLS session from the client is NOT terminated at the Gateway.
-The rest of the document assumes that TLS is being terminated at the Gateway,
-which is the default setting.
+effect as the TLS session from the client is NOT terminated at the Gateway, but rather
+passes through the Gateway, encrypted.
+In the case of `Backend`, only the Upstream connection is affected.
+For `HTTPRoute`, the use of both `Terminate` and `Backend` is supported, and provides what
+is commonly known as a connection that is terminated and then re-encrypted at the Gateway.
```
Comment (robscott) [inline site-src/guides/tls.md:40]: Similar comment as above, I think it's best to revert this change since `Backend` is not really the same thing as `Terminate` or `Passthrough`.

File: site-src/guides/tls.md [hunk 10]
```diff
@@ -98,6 +101,55 @@ would be invalid.
 {% include 'standard/tls-cert-cross-namespace.yaml' %}
 ```
 
+## Upstream TLS
+
+Upstream TLS settings are configured using the experimental `BackendTLSPolicy`
+applied to a `Service` that accesses the backend to be targeted.
+This type of TLS is used when the backend pod has its own certificate and the
+application developer needs to convey how to successfully connect to the
+backend pod from the Gateway.
```
Comment (robscott) [inline site-src/guides/tls.md:110]:
```suggestion

This resource can be used to describe the SNI the Gateway should use to connect to the
backend and how the certificate served by the backend Pod(s) should be verified.
```

File: site-src/guides/tls.md [hunk 11]
```diff
@@ -98,6 +101,55 @@ would be invalid.
 {% include 'standard/tls-cert-cross-namespace.yaml' %}
 ```
 
+## Upstream TLS
+
+Upstream TLS settings are configured using the experimental `BackendTLSPolicy`
+applied to a `Service` that accesses the backend to be targeted.
```
Comment (robscott) [inline site-src/guides/tls.md:107]:
```suggestion
attached to a `Service` via a target reference.
```

File: site-src/guides/tls.md [hunk 12]
```diff
@@ -98,6 +101,55 @@ would be invalid.
...
+The `TargetRef` identifies the `Service` for which your HTTPRoute requires TLS.
+The `TLS` configuration for upstream TLS contains a required `Hostname`, and either
+`CACertRefs` or `WellKnownCACerts`.
+`Hostname` refers to the SNI the Gateway should use to connect to the backend, and
+must match the certificate served by the backend pod.
+`CACertRefs` refer to one or more PEM-encoded TLS certificates.
+If there are no specific certificates to use, then you must set `WellKnownCACerts` to
+"System" to tell the Gateway to use a set of trusted CA Certificates. There may be some
+variation in which system certificates are used by each implementation. Refer to documentation
+from your implementation of choice for more information.
```
Comment (robscott) [inline site-src/guides/tls.md:123]: Would recommend adding some newlines here to split this into more than one paragraph. The deploy preview merges this all into one paragraph that can be a bit hard to follow: https://deploy-preview-2493--kubernetes-sigs-gateway-api.netlify.app/guides/tls/#targetrefs-and-tls

File: site-src/guides/tls.md [hunk 13]
```diff
@@ -98,6 +101,55 @@ would be invalid.
...
+```yaml
+{% include 'experimental/v1alpha2/backendtlspolicy-system-certs.yaml' %}
+```
+
+#### Using Explicit CA Certificates
+
+In this example, the `BackendTLSPolicy` is configured to use certificates defined
+in the configuration map `auth-cert`
+to connect with a TLS-encrypted upstream connection where Pods backing the
+`auth` Service are expected to serve a valid certificate for `auth.example.com`.
```
Comment (robscott) [inline site-src/guides/tls.md:147]: Tiny nit: Line length is a bit inconsistent here.

**Review by robscott — COMMENTED** [review]
Thanks for all the work on this @candita! A few more nits, but mostly LGTM.

**Comment by robscott** [comment]
As a note, I've updated this PR to target `v1-docs` so it can merge as soon as it's ready, no hold necessary. We'll create a separate PR from that branch to the main one to simplify the v1.0 release process.

File: site-src/guides/tls.md (hunk 9 above)
Comment (candita) [reply site-src/guides/tls.md]: Done.

**Comment by candita** [comment]
@robscott I applied your changes, hope this is ready for publication.

**Comment by robscott** [comment]
Thanks @candita!

/lgtm
